### PR TITLE
V0.3.12

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-6021](https://anaconda.atlassian.net/browse/PKG-6021)
- [Upstream repository](https://github.com/langchain-ai/langchain/tree/master/libs/community)
- [Upstream changelog/diff]()

### Explanation of changes:

- Update version
- Sync dependencies with conda forge
- Skip s390x
- Remove Snowflake channel from `abs.yaml`

[PKG-6021]: https://anaconda.atlassian.net/browse/PKG-6021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,101 +21,18 @@ requirements:
...
-    - sqlalchemy >=1.4,<3
+    - langchain <0.4.0,>=0.3.11
+    - langchain-core <0.4.0,>=0.3.24
+    - sqlalchemy <2.0.36,>=1.4
     - requests >=2.0.0,<3.0.0
     - pyyaml >=5.3
     - numpy >=1.0.0,<2.0.0
     - aiohttp >=3.8.3,<4.0.0
     - tenacity >=8.1.0,<9.0.0
     - dataclasses-json >=0.5.7,<0.7
```
Comment (bwingconda) [inline recipe/meta.yaml]: These should come from [here](https://github.com/langchain-ai/langchain/blob/langchain%3D%3D0.3.12/libs/community/pyproject.toml#L31), right?

Comment (SC426) [reply recipe/meta.yaml]: Updated

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,101 +21,18 @@ requirements:
     - pip
   run:
     - python
-    - langchain-core >=0.1.52,<0.2.0
-    - sqlalchemy >=1.4,<3
+    - langchain <0.4.0,>=0.3.11
```
Comment (ELundby45) [inline recipe/meta.yaml]: https://github.com/langchain-ai/langchain/blob/langchain-community%3D%3D0.3.12/libs/community/pyproject.toml#L34

```suggestion
    - langchain <0.4.0,>=0.3.12
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -21,101 +21,18 @@ requirements:
...
+    - langchain-core <0.4.0,>=0.3.24
```
Comment (ELundby45) [inline recipe/meta.yaml]: https://github.com/langchain-ai/langchain/blob/langchain-community%3D%3D0.3.12/libs/community/pyproject.toml#L34

```suggestion
    - langchain-core <0.4.0,>=0.3.25
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -21,101 +21,18 @@ requirements:
...
-    - tidb-vector >=0.0.3,<1.0.0
-    - friendli-client >=1.2.4,<2.0.0
-    - premai >=0.3.25,<0.4.0
-    - vdms >=0.0.20,<0.0.21
-    - httpx-sse >=0.4.0,<0.5.0
-    - pyjwt >=2.8.0,<3.0.0
-    - oracledb >=2.2.0,<3.0.0
+    - pydantic-settings >=2.4.0,<3.0.0
+    - langsmith >=0.1.125,<0.3
+    - httpx-sse >=0.4.0
```
Comment (ELundby45) [inline recipe/meta.yaml]: https://github.com/langchain-ai/langchain/blob/langchain-community%3D%3D0.3.12/libs/community/pyproject.toml#L43

```suggestion
    - httpx-sse >=0.4.0,<0.5.0
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -21,101 +21,18 @@ requirements:
...
   run:
     - python
-    - langchain-core >=0.1.52,<0.2.0
-    - sqlalchemy >=1.4,<3
+    - langchain <0.4.0,>=0.3.11
+    - langchain-core <0.4.0,>=0.3.24
+    - sqlalchemy <3,>=1.4
     - requests >=2.0.0,<3.0.0
     - pyyaml >=5.3
     - numpy >=1.0.0,<2.0.0
```
Comment (ELundby45) [inline recipe/meta.yaml]: https://github.com/langchain-ai/langchain/blob/langchain-community%3D%3D0.3.12/libs/community/pyproject.toml#L44-L50

```suggestion
    - numpy >=1.22.4,<2.0.0  # [py<312]
    - numpy >=1.26.2,<3.0.0  # [py>=312]
```

**Review by ELundby45 — COMMENTED** [review]
Let me know if I am looking in the wrong place?

File: recipe/meta.yaml (hunk 5 above)
Comment (bwingconda) [inline recipe/meta.yaml]:
```suggestion
    - numpy >=1.22.4,<2  # [py<312]
    - numpy >=1.26.2,<3  # [py>=312]
```

**Review by ELundby45 — APPROVED** [review]

**Review by bwingconda — APPROVED** [review]

**Pull request merged**